packages/react-testing/src/element.ts

```diff
@@ -193,26 +197,30 @@ export class Element<Props> implements Node<Props> {
   trigger<K extends FunctionKeys<Props>>(
     prop: K,
     ...args: DeepPartialArguments<Props[K]>
-  ): ReturnType<
-    NonNullable<
-      Props[K] extends ((...args: any[]) => any) | undefined ? Props[K] : never
-    >
-  > {
+  ): NonNullable<Props[K]> extends (...args: any[]) => any
+    ? ReturnType<NonNullable<Props[K]>>
+    : any {
     return this.root.act(() => {
       const propValue = this.props[prop];
 
       if (propValue == null) {
         throw new Error(
-          `Attempted to call prop ${prop} but it was not defined.`,
+          `Attempted to call prop ${String(prop)} but it was not defined.`,
         );
       }
 
       return (propValue as any)(...args);
     });
   }
 
-  triggerKeypath<T = unknown>(keypath: string, ...args: unknown[]): T {
+  triggerKeypath<
+    Path extends string,
+    ExtractedFunction extends KeyPathFunction = ExtractKeypath<Props, Path>,
+  >(
+    ...params: TriggerKeypathParams<Props, Path, ExtractedFunction>
```

Comment:
Having a `keypath` argument here ensured that users must specify at least one argument. By removing the keypath argument, you've made it so `element.triggerKeypath();` is considered valid code from a type checking perspective.

`element.triggerKeypath();` should raise a type error as the first value - the keypath is required. Please retain that behaviour.

Reply:
It does raise an error as expected. `TriggerKeypathParams` is a tuple with a required value in position `0`